orchestrator/cosmos_gravity/src/send.rs

```diff
@@ -120,19 +120,19 @@ pub async fn send_request_batch_tx(
         denom,
     };
     let msg = Msg::new("/gravity.v1.MsgRequestBatchTx", msg_request_batch);
-    send_messages(contact, cosmos_key, fee, vec![msg]).await
+    send_messages(contact, cosmos_key, gas_price, vec![msg]).await
 }
 
 pub async fn send_messages(
     contact: &Contact,
     cosmos_key: CosmosPrivateKey,
-    fee: Coin,
+    gas_price: Coin,
     messages: Vec<Msg>,
 ) -> Result<TxResponse, CosmosGrpcError> {
     let cosmos_address = cosmos_key.to_address(&contact.get_prefix()).unwrap();
 
     let fee = Fee {
-        amount: vec![fee],
+        amount: vec![gas_price],
```

Comment:
Recall, `fees = ceil(gasPrice * gasLimit)`. So, I don't think we want to provide `gas_price` here for the `amount`. Instead, I think we need to compute the actual fee. i.e. do the following:

```rust
let gas_limit = 500_000_000u64 * (messages.len() as u64);

let fee = Fee {
  amount: vec![ceil(gas_limit * gas_price)],
  gas_limit: gas_limit,
  // ...
}
```

You can just translate the following [SDK code](https://github.com/cosmos/cosmos-sdk/blob/master/client/tx/factory.go#L207-L222) to Rust.

Reply:
That makes a lot more sense. Will push it up shortly

Reply:
Forgive my incorrect Rust :p

Reply:
No worries, the basic formula for computing fees is getting lost in a sea of types here. Currently trying to understand how `Coin` wants me to multiply & find a ceiling, lol

Reply:
@levicook, just take the amount. Again, incorrect rust:

```rust
let fee = Coin{denom: gas_price.Denom, amount: ceil(gas_price.Amount * gas_limit)}
```